src/main/java/org/jenkinsci/plugins/deployment/DeploymentBuildWrapper.java

```diff
@@ -0,0 +1,67 @@
+package org.jenkinsci.plugins.deployment;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildWrapper;
+import hudson.tasks.BuildWrapperDescriptor;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class DeploymentBuildWrapper extends BuildWrapper {
+    @DataBoundConstructor
+    public DeploymentBuildWrapper() {
+    }
+
+    @Override
+    public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener listener) {
+        final EnvVars env = new EnvVars();
+        final List<String> deploymentHost = new ArrayList();
+        final List<String> deploymentPath = new ArrayList();
+        final List<String> deploymentTimeStamp = new ArrayList();
+
+        final HostRecords hostRecords = build.getAction(HostRecords.class);
+
+        return new Environment() {
+            @Override
+            public void buildEnvVars(Map<String, String> env) {
+                if(hostRecords !=null) {
+                    Collection<HostRecord> hostRecordList = hostRecords.getHostrecords();
+                    Iterator hostRecordListIterator = hostRecordList.iterator();
+                    while(hostRecordListIterator.hasNext()) {
```

Comment:
Minor: <code>for (HostRecored hostRecord : hostRecords)</code> would simplify the code. It applies to other cycles as well